src/App.tsx

```diff
@@ -1,17 +1,94 @@
-import React from 'react';
-import './App.css';
-
+import React, { ChangeEvent, useState } from 'react';
+import './App.scss';
 import users from './api/users';
+import todos from './api/todos';
+import { PreparedTodo } from './types/app.typedefs';
+import { TodoList } from './components/TodoList/TodoList';
+
+const preparedTodos: PreparedTodo[] = todos.map((todo) => {
+  const user = users.find(person => person.id === todo.userId) || undefined;
+
+  return (
+    {
+      ...todo,
+      user,
+    }
+  );
+});
 
 const App: React.FC = () => {
+  const [todoList, setTodoList] = useState(preparedTodos);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [todoUser, setTodoUser] = useState(0);
+  const [showTitleError, setShowTitleError] = useState(false);
+  const [showUserError, setShowUserError] = useState(false);
+
+  const addTodo = (title: string, userId: number):void => {
+    const todo: PreparedTodo = {
+      userId,
+      id: todoList.length + 1,
+      title,
+      completed: false,
+      user: users.find(user => user.id === userId),
+    };
+
+    setTodoList([...todoList, todo]);
+  };
+
+  const handleSubmit = (event: React.FormEvent):void => {
+    event.preventDefault();
+
+    if (todoTitle && todoUser) {
+      addTodo(todoTitle, todoUser);
+      setTodoTitle('');
+      setTodoUser(0);
+    }
+
+    setShowTitleError(!todoTitle);
+    setShowUserError(!todoUser);
+  };
+
   return (
     <div className="App">
-      <h1>Add todo form</h1>
+      <form onSubmit={handleSubmit}>
+        <input
+          type="text"
+          data-cy="titleInput"
```

Comment:
Using labels is good practice lets use it!

Reply:
yes, labels should serve their purpose - namely if we need to expand the clickable area,
yet in @LukaszGryziewicz case, following the minimal styling there's no real need for a label - it's only an input that you focus at.
So not an actual blocking matter.